Adapt and enhance the releasing tools

Desvelao opened the pull request:
### Description
This pull request adapts and enhances the releasing tools to manage the plugin for the Wazuh dashboard platform.

Changes:
- Create tag and bump script to manage the repository
- Adapt and enhance the RELEASING.md file
- Remove unused releasing tools in plugins/main plugin
- Support for bumping multi plugins
- Support for running a specific bump task for each plugin
 
### Issues Resolved
#6091 

### Evidence

<details>

<summary>tag script - help</summary>

```console
$ node scripts/release/tag.js
tag - Help
Bump the plugins and create the tag in the remote repository.
Some warning messages are sent to stderr.

Usage: node scripts/release/tag.js [options]

Options:
--debug
  Enable debug in the logger.
--display-configuration
  Display the configuration.
--display-examples
  Display the examples.
--help
  Display the help.
--version <version>
  Set the version.
--revision <revision>
  Set the revision.
--platform-version <platform-version>
  Set the platform version.
--ignore-confirmation
  Ignore the confirmation.
--manifest-package <file-path>
  Set the package manifest file location to take the default values.
--tag-suffix <suffix>
  Set the tag suffix
  ```

</details>

<details>

<summary>bump script - help</summary>

```console
bump - Help
Bump the plugin version, revision and/or platform version
Some warning messages are sent to stderr.

Usage: node /home/toni/wazuh/src/wazuh-kibana-app/.worktrees/triage/scripts/release/bump.js [options]

Options:
--debug
  Enable debug in the logger.
--display-configuration
  Display the configuration.
--help
  Display the help.
--version <version>
  Set the version.
--revision <revision>
  Set the revision.
--platform-version <platform-version>
  Set the platform version.
--ignore-confirmation
  Ignore the confirmation.
--manifest-changelog <suffix>
  Set the tag suffix
--manifest-package <file-path>
  Set the package manifest file location to take the default values.
--plugins-directory <directory>
  Set the plugins directory where they are located.
```

</details>

### Test
See RELEASING.md.

Legend:
:black_circle:: none
:green_circle:: pass
:yellow_circle:: warning
:red_circle:: fail
:white_circle:: not applicable

## Other

| Test | Result |
| --- |  --- |
| Run the tag script is able to create and push the tag to the remote | :black_circle: |
| Run the bump script is able to bump the plugins and changelog file | :black_circle: |

**Details**
<details>
<summary>:black_circle: Run the tag script is able to create and push the tag to the remote</summary>

</details>

<details>
<summary>:black_circle: Run the bump script is able to bump the plugins and changelog file</summary>

</details>


### Check List
- [ ] All tests pass
  - [ ] `yarn test:jest`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

yenienserrano reviewed (approved):
CR: ✅
Test: 🔴 

I think we should update the examples of use of the scripts since tags.sh and bump.sh are failing.


<details><summary>logs</summary>

```console
fork ➤ node scripts/release/tag.js --plugins-directory ./plugins --version 4.7.1 --revision 05 --platform-version 2.8.0                                                                                    git:enhancement/6091-adapt-enhance-releasing-tools
[INFO][tag]: Version: 4.7.1
[INFO][tag]: Revision: 05
[INFO][tag]: Platform version: 2.8.0
[WARN][tag]: Ensure the base branch is created in the remote and it has updated the files: CHANGELOG.md, unit tests files, API data files, etc... See RELEASING.md. It does not modify these files.
[WARN][tag]: This script will prune the local branches and tags, bump the plugins, commit and push the tags to the remote repository, deleting any unpushed changes.
Do you want to continue? [y/N] y
[INFO][tag]: Run command: git checkout 4.7.1
Switched to a new branch '4.7.1'
[INFO][tag]: Switched to branch: 4.7.1
[INFO][tag]: Run command: git fetch --prune --prune-tags --force
From https://github.com/wazuh/wazuh-kibana-app
 - [deleted]             (none)     -> v4.5.0-2.6.0-test
 - [deleted]             (none)     -> v4.5.0-2.6.0-test1
 - [deleted]             (none)     -> v4.5.0-7.10.2-test1
 - [deleted]             (none)     -> v4.5.0-7.16.0-test1
 - [deleted]             (none)     -> v4.5.0-7.16.1-test1
 - [deleted]             (none)     -> v4.5.0-7.16.2-test1
 - [deleted]             (none)     -> v4.5.0-7.16.3-test1
 - [deleted]             (none)     -> v4.5.0-7.17.0-test1
 - [deleted]             (none)     -> v4.5.0-7.17.1-test1
 - [deleted]             (none)     -> v4.5.0-7.17.10-test1
 - [deleted]             (none)     -> v4.5.0-7.17.2-test1
 - [deleted]             (none)     -> v4.5.0-7.17.3-test1
 - [deleted]             (none)     -> v4.5.0-7.17.4-test1
 - [deleted]             (none)     -> v4.5.0-7.17.5-test1
 - [deleted]             (none)     -> v4.5.0-7.17.6-test1
 - [deleted]             (none)     -> v4.5.0-7.17.7-test1
 - [deleted]             (none)     -> v4.5.0-7.17.8-test1
 - [deleted]             (none)     -> v4.5.0-7.17.9-test1
 - [deleted]             (none)     -> v4.6.0-7.17.9-pre-alpha1
 - [deleted]             (none)     -> v4.6.0-pre-alpha1
[INFO][tag]: Pruned local branches and tags
[INFO][tag]: Generating tag: v4.7.1-2.8.0...
[INFO][bump]: Plugin directories from ./plugins: main
[INFO][bump][main]: Version: 4.7.1
[INFO][bump][main]: Revision: 05
[INFO][bump][main]: Platform version: 2.8.0
[INFO][bump][main]: Package manifest: /home/ian/wazuh/kibana/fork/plugins/main/package.json
[INFO][bump][main]: Plugin manifest: /home/ian/wazuh/kibana/fork/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: This will update the manifest files: package and platform plugin.
[INFO][bump][main]: Updated /home/ian/wazuh/kibana/fork/plugins/main/package.json
[INFO][bump][main]: Updated /home/ian/wazuh/kibana/fork/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: Calling to specific bump script
[INFO]: Version: 4.7.1
[INFO]: Revision: 05
[INFO]: Platform version: 2.8.0
[INFO]: Package manifest: /home/ian/wazuh/kibana/fork/plugins/main/package.json
[INFO]: Plugin manifest: /home/ian/wazuh/kibana/fork/plugins/main/opensearch_dashboards.json
[INFO]: This will update the manifest files: package and platform plugin.
[INFO]: Updated /home/ian/wazuh/kibana/fork/plugins/main/package.json
[INFO]: Updated /home/ian/wazuh/kibana/fork/plugins/main/opensearch_dashboards.json
[ERROR]: changelog file is not defined. Use --manifest-changelog <path/to/file>.
```

```console
wazuh-dashboard-plugins ➤ node scripts/release/bump.js --plugins-directory ./plugins --version 4.6.0 --revision 05 --platform-version 2.11.0                                                              git:enhancement/6091-adapt-enhance-releasing-tools*
[INFO][bump]: Plugin directories from ./plugins: main, wazuh-check-updates, wazuh-core
[INFO][bump][main]: Version: 4.6.0
[INFO][bump][main]: Revision: 05
[INFO][bump][main]: Platform version: 2.11.0
[INFO][bump][main]: Package manifest: /home/ian/wazuh/kibana/wazuh-dashboard-plugins/plugins/main/package.json
[INFO][bump][main]: Plugin manifest: /home/ian/wazuh/kibana/wazuh-dashboard-plugins/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: This will update the manifest files: package and platform plugin.
[INFO][bump][main]: Updated /home/ian/wazuh/kibana/wazuh-dashboard-plugins/plugins/main/package.json
[INFO][bump][main]: Updated /home/ian/wazuh/kibana/wazuh-dashboard-plugins/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: Calling to specific bump script
[ERROR][bump][main][specific]: This plugin must update the API data. Use --plugin-main-generate-api-data-spec <plugin-main-generate-api-data-spec> to define the spec URL file.
```

</details>

yenienserrano reviewed (commented):
Comment: https://github.com/wazuh/wazuh-dashboard-plugins/pull/6092#pullrequestreview-1717988381

yenienserrano commented:
<details><summary>Test bump</summary>

```console
fork ➤ node scripts/release/bump.js --plugins-directory ./plugins --manifest-changelog ./CHANGELOG.md --plugin-main-generate-api-data-spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml --version 4.6.0 --revision 03 --platform-version 2.8.0
[INFO][bump]: Plugin directories from ./plugins: main
[INFO][bump][main]: Version: 4.6.0
[INFO][bump][main]: Revision: 03
[INFO][bump][main]: Platform version: 2.8.0
[INFO][bump][main]: Package manifest: /home/ian/wazuh/kibana/fork/plugins/main/package.json
[INFO][bump][main]: Plugin manifest: /home/ian/wazuh/kibana/fork/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: This will update the manifest files: package and platform plugin.
[INFO][bump][main]: Updated /home/ian/wazuh/kibana/fork/plugins/main/package.json
[INFO][bump][main]: Updated /home/ian/wazuh/kibana/fork/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: Calling to specific bump script
[INFO][bump][main][specific]: Run command: cd /home/ian/wazuh/kibana/fork/plugins/main && yarn generate:api-data --spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml
{
  displayConfiguration: true,
  displayExamples: false,
  displayHelp: false,
  format: 'plugin',
  output: 'file',
  spec: 'https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml',
  outputDirectory: '/home/ian/wazuh/kibana/fork/plugins/main/common/api-info'
}
yarn run v1.22.19
$ node scripts/generate-api-data.js --spec https://raw.githubusercontent.com/wazuh/wazuh/$(node -e "console.log(require('./package.json').version)")/api/api/spec/spec.yaml --output file --output-directory common/api-info --display-configuration --spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml
File was created! Path: /home/ian/wazuh/kibana/fork/plugins/main/common/api-info/security-actions.json
File was created! Path: /home/ian/wazuh/kibana/fork/plugins/main/common/api-info/endpoints.json
Done in 1.62s.

fork ➤ node scripts/release/bump.js --plugins-directory ./plugins --manifest-changelog ./CHANGELOG.md --plugin-main-generate-api-data-spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml --manifest-package ./plugins/main/package.json --revision 03
[WARN][bump]: version is not defined. Using from the current package manifest ./plugins/main/package.json: 4.7.0
[INFO][bump]: Plugin directories from ./plugins: main
[INFO][bump][main]: Version: 4.7.0
[INFO][bump][main]: Revision: 03
[INFO][bump][main]: Package manifest: /home/ian/wazuh/kibana/fork/plugins/main/package.json
[INFO][bump][main]: Plugin manifest: /home/ian/wazuh/kibana/fork/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: This will update the manifest files: package and platform plugin.
[INFO][bump][main]: Updated /home/ian/wazuh/kibana/fork/plugins/main/package.json
[INFO][bump][main]: Updated /home/ian/wazuh/kibana/fork/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: Calling to specific bump script
[INFO][bump][main][specific]: Run command: cd /home/ian/wazuh/kibana/fork/plugins/main && yarn generate:api-data --spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml
{
  displayConfiguration: true,
  displayExamples: false,
  displayHelp: false,
  format: 'plugin',
  output: 'file',
  spec: 'https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml',
  outputDirectory: '/home/ian/wazuh/kibana/fork/plugins/main/common/api-info'
}
yarn run v1.22.19
$ node scripts/generate-api-data.js --spec https://raw.githubusercontent.com/wazuh/wazuh/$(node -e "console.log(require('./package.json').version)")/api/api/spec/spec.yaml --output file --output-directory common/api-info --display-configuration --spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml
File was created! Path: /home/ian/wazuh/kibana/fork/plugins/main/common/api-info/security-actions.json
File was created! Path: /home/ian/wazuh/kibana/fork/plugins/main/common/api-info/endpoints.json
Done in 1.64s.

[INFO][bump]: Updated ./CHANGELOG.md
fork ➤ node scripts/release/bump.js --plugins-directory ./plugins --manifest-changelog ./CHANGELOG.md --plugin-main-generate-api-data-spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml --manifest-package ./plugins/main/package.json --version 4.6.0
[WARN][bump]: revision is not defined. Using from the current package manifest ./plugins/main/package.json: 02
[INFO][bump]: Plugin directories from ./plugins: main
[INFO][bump][main]: Version: 4.6.0
[INFO][bump][main]: Revision: 02
[INFO][bump][main]: Package manifest: /home/ian/wazuh/kibana/fork/plugins/main/package.json
[INFO][bump][main]: Plugin manifest: /home/ian/wazuh/kibana/fork/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: This will update the manifest files: package and platform plugin.
[INFO][bump][main]: Updated /home/ian/wazuh/kibana/fork/plugins/main/package.json
[INFO][bump][main]: Updated /home/ian/wazuh/kibana/fork/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: Calling to specific bump script
[INFO][bump][main][specific]: Run command: cd /home/ian/wazuh/kibana/fork/plugins/main && yarn generate:api-data --spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml
{
  displayConfiguration: true,
  displayExamples: false,
  displayHelp: false,
  format: 'plugin',
  output: 'file',
  spec: 'https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml',
  outputDirectory: '/home/ian/wazuh/kibana/fork/plugins/main/common/api-info'
}
yarn run v1.22.19
$ node scripts/generate-api-data.js --spec https://raw.githubusercontent.com/wazuh/wazuh/$(node -e "console.log(require('./package.json').version)")/api/api/spec/spec.yaml --output file --output-directory common/api-info --display-configuration --spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml
File was created! Path: /home/ian/wazuh/kibana/fork/plugins/main/common/api-info/security-actions.json
File was created! Path: /home/ian/wazuh/kibana/fork/plugins/main/common/api-info/endpoints.json
Done in 1.35s.

[INFO][bump]: Updated ./CHANGELOG.md
fork ➤ node scripts/release/bump.js --plugins-directory ./plugins --manifest-changelog ./CHANGELOG.md --plugin-main-generate-api-data-spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml --manifest-package ./plugins/main/package.json --platform-version 2.9.0
[WARN][bump]: version is not defined. Using from the current package manifest ./plugins/main/package.json: 4.7.0
[WARN][bump]: revision is not defined. Using from the current package manifest ./plugins/main/package.json: 02
[INFO][bump]: Plugin directories from ./plugins: main
[INFO][bump][main]: Version: 4.7.0
[INFO][bump][main]: Revision: 02
[INFO][bump][main]: Platform version: 2.9.0
[INFO][bump][main]: Package manifest: /home/ian/wazuh/kibana/fork/plugins/main/package.json
[INFO][bump][main]: Plugin manifest: /home/ian/wazuh/kibana/fork/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: This will update the manifest files: package and platform plugin.
[INFO][bump][main]: Updated /home/ian/wazuh/kibana/fork/plugins/main/package.json
[INFO][bump][main]: Updated /home/ian/wazuh/kibana/fork/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: Calling to specific bump script
[INFO][bump][main][specific]: Run command: cd /home/ian/wazuh/kibana/fork/plugins/main && yarn generate:api-data --spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml
{
  displayConfiguration: true,
  displayExamples: false,
  displayHelp: false,
  format: 'plugin',
  output: 'file',
  spec: 'https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml',
  outputDirectory: '/home/ian/wazuh/kibana/fork/plugins/main/common/api-info'
}
yarn run v1.22.19
$ node scripts/generate-api-data.js --spec https://raw.githubusercontent.com/wazuh/wazuh/$(node -e "console.log(require('./package.json').version)")/api/api/spec/spec.yaml --output file --output-directory common/api-info --display-configuration --spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml
File was created! Path: /home/ian/wazuh/kibana/fork/plugins/main/common/api-info/security-actions.json
File was created! Path: /home/ian/wazuh/kibana/fork/plugins/main/common/api-info/endpoints.json
Done in 1.61s.

[INFO][bump]: Updated ./CHANGELOG.md
fork ➤ node scripts/release/bump.js --plugins-directory ./plugins --manifest-changelog ./CHANGELOG.md --plugin-main-generate-api-data-spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml --version 4.6.0 --revision 03 --platform-version 2.8.0
[INFO][bump]: Plugin directories from ./plugins: main
[INFO][bump][main]: Version: 4.6.0
[INFO][bump][main]: Revision: 03
[INFO][bump][main]: Platform version: 2.8.0
[INFO][bump][main]: Package manifest: /home/ian/wazuh/kibana/fork/plugins/main/package.json
[INFO][bump][main]: Plugin manifest: /home/ian/wazuh/kibana/fork/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: This will update the manifest files: package and platform plugin.
[INFO][bump][main]: Updated /home/ian/wazuh/kibana/fork/plugins/main/package.json
[INFO][bump][main]: Updated /home/ian/wazuh/kibana/fork/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: Calling to specific bump script
[INFO][bump][main][specific]: Run command: cd /home/ian/wazuh/kibana/fork/plugins/main && yarn generate:api-data --spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml
{
  displayConfiguration: true,
  displayExamples: false,
  displayHelp: false,
  format: 'plugin',
  output: 'file',
  spec: 'https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml',
  outputDirectory: '/home/ian/wazuh/kibana/fork/plugins/main/common/api-info'
}
yarn run v1.22.19
$ node scripts/generate-api-data.js --spec https://raw.githubusercontent.com/wazuh/wazuh/$(node -e "console.log(require('./package.json').version)")/api/api/spec/spec.yaml --output file --output-directory common/api-info --display-configuration --spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml
File was created! Path: /home/ian/wazuh/kibana/fork/plugins/main/common/api-info/security-actions.json
File was created! Path: /home/ian/wazuh/kibana/fork/plugins/main/common/api-info/endpoints.json
Done in 1.49s.
```

</details>

yenienserrano commented:
<details><summary>Test tags ✅ </summary>

```console
fork ➤ node scripts/release/tag.js --plugins-directory ./plugins --manifest-changelog ./CHANGELOG.md --plugin-main-generate-api-data-spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml --version 4.6.0 --revision 05 --platform-version 2.10.0
[INFO][tag]: Version: 4.6.0
[INFO][tag]: Revision: 05
[INFO][tag]: Platform version: 2.10.0
[WARN][tag]: Ensure the base branch is created in the remote and it has updated the files: CHANGELOG.md, unit tests files, API data files, etc... See RELEASING.md. It does not modify these files.
[WARN][tag]: This script will prune the local branches and tags, bump the plugins, commit and push the tags to the remote repository, deleting any unpushed changes.
Do you want to continue? [y/N] y
[INFO][tag]: Run command: git checkout 4.6.0
Already on '4.6.0'
[INFO][tag]: Switched to branch: 4.6.0
[INFO][tag]: Run command: git fetch --prune --prune-tags --force
[INFO][tag]: Pruned local branches and tags
[INFO][tag]: Generating tag: v4.6.0-2.10.0...
[INFO][bump]: Plugin directories from ./plugins: main
[INFO][bump][main]: Version: 4.6.0
[INFO][bump][main]: Revision: 05
[INFO][bump][main]: Platform version: 2.10.0
[INFO][bump][main]: Package manifest: /home/ian/wazuh/kibana/fork/plugins/main/package.json
[INFO][bump][main]: Plugin manifest: /home/ian/wazuh/kibana/fork/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: This will update the manifest files: package and platform plugin.
[INFO][bump][main]: Updated /home/ian/wazuh/kibana/fork/plugins/main/package.json
[INFO][bump][main]: Updated /home/ian/wazuh/kibana/fork/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: Calling to specific bump script
[INFO][bump][main][specific]: Run command: cd /home/ian/wazuh/kibana/fork/plugins/main && yarn generate:api-data --spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml
{
  displayConfiguration: true,
  displayExamples: false,
  displayHelp: false,
  format: 'plugin',
  output: 'file',
  spec: 'https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml',
  outputDirectory: '/home/ian/wazuh/kibana/fork/plugins/main/common/api-info'
}
yarn run v1.22.19
$ node scripts/generate-api-data.js --spec https://raw.githubusercontent.com/wazuh/wazuh/$(node -e "console.log(require('./package.json').version)")/api/api/spec/spec.yaml --output file --output-directory common/api-info --display-configuration --spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml
File was created! Path: /home/ian/wazuh/kibana/fork/plugins/main/common/api-info/security-actions.json
File was created! Path: /home/ian/wazuh/kibana/fork/plugins/main/common/api-info/endpoints.json
Done in 1.76s.

[INFO][bump]: Updated ./CHANGELOG.md
[INFO][tag]: Run command: git diff --exit-code --no-patch;echo -n $?
Review the changes done with the bump. y (continue), d (list changes: git diff), f (list modified files: git diff --name-only), N/n (cancel) f
[INFO][tag]: Run command: git diff --name-only
CHANGELOG.md
plugins/main/common/api-info/endpoints.json
plugins/main/common/api-info/security-actions.json
plugins/main/opensearch_dashboards.json
plugins/main/package.json

Review the changes done with the bump. y (continue), d (list changes: git diff), f (list modified files: git diff --name-only), N/n (cancel) y
[INFO][tag]: There are changes to commit.
[INFO][tag]: Run command: git diff --name-only
CHANGELOG.md
plugins/main/common/api-info/endpoints.json
plugins/main/common/api-info/security-actions.json
plugins/main/opensearch_dashboards.json
plugins/main/package.json

[INFO][tag]: Run command: git commit -am "Bump v4.6.0-2.10.0"
[INFO][tag]: Commited
[INFO][tag]: Run command: git tag -a v4.6.0-2.10.0 -m "Wazuh plugins for Wazuh dashboard v4.6.0-2.10.0"
[INFO][tag]: Created tag: v4.6.0-2.10.0
[INFO][tag]: Run command: git push origin v4.6.0-2.10.0
To https://github.com/yenienserrano/wazuh-kibana-app
 * [new tag]             v4.6.0-2.10.0 -> v4.6.0-2.10.0
[INFO][tag]: Pushed tag v4.6.0-2.10.0 to remote
[INFO][tag]: Run command: git reset --hard origin/4.6.0
[INFO][tag]: Undone changes.
[WARN][tag]: A commit was added to the tag, but this was removed in the 4.6.0.
```

</details>

yenienserrano reviewed (approved):
LGTM

JuanGarriuz commented:
<details><summary>Test bump ✅ </summary>

```console
node scripts/release/bump.js --plugins-directory ./plugins --manifest-changelog ./CHANGELOG.md --plugin-main-generate-api-data-spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml --version 4.7.0 --revision 07 --platform-version 2.8.0
[INFO][bump]: Plugin directories from ./plugins: main
[INFO][bump][main]: Version: 4.7.0
[INFO][bump][main]: Revision: 07
[INFO][bump][main]: Platform version: 2.8.0
[INFO][bump][main]: Package manifest: /home/usuario/wazuh-kibana-app/plugins/main/package.json
[INFO][bump][main]: Plugin manifest: /home/usuario/wazuh-kibana-app/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: This will update the manifest files: package and platform plugin.
[INFO][bump][main]: Updated /home/usuario/wazuh-kibana-app/plugins/main/package.json
[INFO][bump][main]: Updated /home/usuario/wazuh-kibana-app/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: Calling to specific bump script
[INFO][bump][main][specific]: Run command: cd /home/usuario/wazuh-kibana-app/plugins/main && yarn generate:api-data --spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml
{
  displayConfiguration: true,
  displayExamples: false,
  displayHelp: false,
  format: 'plugin',
  output: 'file',
  spec: 'https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml',
  outputDirectory: '/home/usuario/wazuh-kibana-app/plugins/main/common/api-info'
}
yarn run v1.22.19
$ node scripts/generate-api-data.js --spec https://raw.githubusercontent.com/wazuh/wazuh/$(node -e "console.log(require('./package.json').version)")/api/api/spec/spec.yaml --output file --output-directory common/api-info --display-configuration --spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml
File was created! Path: /home/usuario/wazuh-kibana-app/plugins/main/common/api-info/security-actions.json
File was created! Path: /home/usuario/wazuh-kibana-app/plugins/main/common/api-info/endpoints.json
Done in 2.36s.

[INFO][bump]: Updated ./CHANGELOG.md

Tostti commented:
### :red_circle: Test tag

The test fails telling that the changelog file is missing, suggesting to use `--manifest-changelog` to add it

<details>
```
 node scripts/release/tag.js --plugins-directory ./plugins --version 4.7.1 --revision 02 --platform-version 2.9.0
[INFO][tag]: Version: 4.7.1
[INFO][tag]: Revision: 02
[INFO][tag]: Platform version: 2.9.0
[WARN][tag]: Ensure the base branch is created in the remote and it has updated the files: CHANGELOG.md, unit tests files, API data files, etc... See RELEASING.md. It does not modify these files.
[WARN][tag]: This script will prune the local branches and tags, bump the plugins, commit and push the tags to the remote repository, deleting any unpushed changes.
Do you want to continue? [y/N] y
[INFO][tag]: Run command: git checkout 4.7.1
Switched to a new branch '4.7.1'
[INFO][tag]: Switched to branch: 4.7.1
[INFO][tag]: Run command: git fetch --prune --prune-tags --force
[INFO][tag]: Pruned local branches and tags
[INFO][tag]: Generating tag: v4.7.1-2.9.0...
[INFO][bump]: Plugin directories from ./plugins: main
[INFO][bump][main]: Version: 4.7.1
[INFO][bump][main]: Revision: 02
[INFO][bump][main]: Platform version: 2.9.0
[INFO][bump][main]: Package manifest: /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/package.json
[INFO][bump][main]: Plugin manifest: /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: This will update the manifest files: package and platform plugin.
[INFO][bump][main]: Updated /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/package.json
[INFO][bump][main]: Updated /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: Calling to specific bump script
[INFO]: Version: 4.7.1
[INFO]: Revision: 02
[INFO]: Platform version: 2.9.0
[INFO]: Package manifest: /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/package.json
[INFO]: Plugin manifest: /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/opensearch_dashboards.json
[INFO]: This will update the manifest files: package and platform plugin.
[INFO]: Updated /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/package.json
[INFO]: Updated /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/opensearch_dashboards.json
[ERROR]: changelog file is not defined. Use --manifest-changelog <path/to/file>.
```
</details>

However, applying that option the error persists

<details>

```
❯ node scripts/release/tag.js --plugins-directory ./plugins --manifest-changelog ./CHANGELOG.md   --version 4.7.1 --revision 02 --platform-version 2.9.0

[INFO][tag]: Version: 4.7.1
[INFO][tag]: Revision: 02
[INFO][tag]: Platform version: 2.9.0
[WARN][tag]: Ensure the base branch is created in the remote and it has updated the files: CHANGELOG.md, unit tests files, API data files, etc... See RELEASING.md. It does not modify these files.
[WARN][tag]: This script will prune the local branches and tags, bump the plugins, commit and push the tags to the remote repository, deleting any unpushed changes.
Do you want to continue? [y/N] y
[INFO][tag]: Run command: git checkout 4.7.1
Switched to branch '4.7.1'
[INFO][tag]: Switched to branch: 4.7.1
[INFO][tag]: Run command: git fetch --prune --prune-tags --force
[INFO][tag]: Pruned local branches and tags
[INFO][tag]: Generating tag: v4.7.1-2.9.0...
[INFO][bump]: Plugin directories from ./plugins: main
[INFO][bump][main]: Version: 4.7.1
[INFO][bump][main]: Revision: 02
[INFO][bump][main]: Platform version: 2.9.0
[INFO][bump][main]: Package manifest: /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/package.json
[INFO][bump][main]: Plugin manifest: /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: This will update the manifest files: package and platform plugin.
[INFO][bump][main]: Updated /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/package.json
[INFO][bump][main]: Updated /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: Calling to specific bump script
[INFO]: Version: 4.7.1
[INFO]: Revision: 02
[INFO]: Platform version: 2.9.0
[INFO]: Package manifest: /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/package.json
[INFO]: Plugin manifest: /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/opensearch_dashboards.json
[INFO]: This will update the manifest files: package and platform plugin.
[INFO]: Updated /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/package.json
[INFO]: Updated /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/opensearch_dashboards.json
[ERROR]: changelog file is not defined. Use --manifest-changelog <path/to/file>.

```

</details>

If I look for the script help, `--manifest-changelog` is not listed

<details>

```
❯ node scripts/release/tag.js --help

tag - Help
Bump the plugins and create the tag in the remote repository.
Some warning messages are sent to stderr.

Usage: node /home/tostti/src/forks/wazuh-dashboard-plugins/scripts/release/tag.js [options]

Options:
--debug
  Enable debug in the logger.
--display-configuration
  Display the configuration.
--display-examples
  Display the examples.
--help
  Display the help.
--version <version>
  Set the version.
--revision <revision>
  Set the revision.
--platform-version <platform-version>
  Set the platform version.
--ignore-confirmation
  Ignore the confirmation.
--manifest-package <file-path>
  Set the package manifest file location to take the default values.
--tag-suffix <suffix>
  Set the tag suffix
```

</details>

### 🟡 : Test bump

The script works 

<details>

```
node scripts/release/bump.js --plugins-directory ./plugins --manifest-changelog ./CHANGELOG.md --plugin-main-generate-api-data-spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml --version 4.6.0 --revision 01 --platform-version 2.8.0
[INFO][bump]: Plugin directories from ./plugins: main
[INFO][bump][main]: Version: 4.6.0
[INFO][bump][main]: Revision: 01
[INFO][bump][main]: Platform version: 2.8.0
[INFO][bump][main]: Package manifest: /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/package.json
[INFO][bump][main]: Plugin manifest: /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: This will update the manifest files: package and platform plugin.
[INFO][bump][main]: Updated /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/package.json
[INFO][bump][main]: Updated /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: Calling to specific bump script
[INFO][bump][main][specific]: Run command: cd /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main && yarn generate:api-data --spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml
{
  displayConfiguration: true,
  displayExamples: false,
  displayHelp: false,
  format: 'plugin',
  output: 'file',
  spec: 'https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml',
  outputDirectory: '/home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/common/api-info'
}
yarn run v1.22.19
$ node scripts/generate-api-data.js --spec https://raw.githubusercontent.com/wazuh/wazuh/$(node -e "console.log(require('./package.json').version)")/api/api/spec/spec.yaml --output file --output-directory common/api-info --display-configuration --spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml
File was created! Path: /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/common/api-info/security-actions.json
File was created! Path: /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/common/api-info/endpoints.json
Done in 3.57s.

[INFO][bump]: Updated ./CHANGELOG.md

```

</details>

The help shows incorrect information for the `--manifest-changelog` option

<details>

```
❯ node scripts/release/bump.js --help
bump - Help
Bump the plugin version, revision and/or platform version
Some warning messages are sent to stderr.

Usage: node /home/tostti/src/forks/wazuh-dashboard-plugins/scripts/release/bump.js [options]

Options:
--debug
  Enable debug in the logger.
--display-configuration
  Display the configuration.
--help
  Display the help.
--version <version>
  Set the version.
--revision <revision>
  Set the revision.
--platform-version <platform-version>
  Set the platform version.
--ignore-confirmation
  Ignore the confirmation.
--manifest-changelog <suffix>
  Set the tag suffix
--manifest-package <file-path>
  Set the package manifest file location to take the default values.
--plugins-directory <directory>
  Set the plugins directory where they are located.

```

</details>

Desvelao commented:
> If I look for the script help, --manifest-changelog is not listed

Yes, you are right. This is caused because the option `--manifest-changelog` is not used by the script tag itself, else by the `bump` process that is wrapped by the `tag` process. The same problem happens with other options.

For now, I will add the option of bump process to tag, but I don't like this solution due to the duplication. I was thinking about the `tag` CLI inheriting from the `bump`, so the duplication could be avoided.

Desvelao commented:
> ### 🔴 Test tag
> The test fails telling that the changelog file is missing, suggesting to use `--manifest-changelog` to add it
> 
> ```
>  node scripts/release/tag.js --plugins-directory ./plugins --version 4.7.1 --revision 02 --platform-version 2.9.0
> [INFO][tag]: Version: 4.7.1
> [INFO][tag]: Revision: 02
> [INFO][tag]: Platform version: 2.9.0
> [WARN][tag]: Ensure the base branch is created in the remote and it has updated the files: CHANGELOG.md, unit tests files, API data files, etc... See RELEASING.md. It does not modify these files.
> [WARN][tag]: This script will prune the local branches and tags, bump the plugins, commit and push the tags to the remote repository, deleting any unpushed changes.
> Do you want to continue? [y/N] y
> [INFO][tag]: Run command: git checkout 4.7.1
> Switched to a new branch '4.7.1'
> [INFO][tag]: Switched to branch: 4.7.1
> [INFO][tag]: Run command: git fetch --prune --prune-tags --force
> [INFO][tag]: Pruned local branches and tags
> [INFO][tag]: Generating tag: v4.7.1-2.9.0...
> [INFO][bump]: Plugin directories from ./plugins: main
> [INFO][bump][main]: Version: 4.7.1
> [INFO][bump][main]: Revision: 02
> [INFO][bump][main]: Platform version: 2.9.0
> [INFO][bump][main]: Package manifest: /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/package.json
> [INFO][bump][main]: Plugin manifest: /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/opensearch_dashboards.json
> [INFO][bump][main]: This will update the manifest files: package and platform plugin.
> [INFO][bump][main]: Updated /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/package.json
> [INFO][bump][main]: Updated /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/opensearch_dashboards.json
> [INFO][bump][main]: Calling to specific bump script
> [INFO]: Version: 4.7.1
> [INFO]: Revision: 02
> [INFO]: Platform version: 2.9.0
> [INFO]: Package manifest: /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/package.json
> [INFO]: Plugin manifest: /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/opensearch_dashboards.json
> [INFO]: This will update the manifest files: package and platform plugin.
> [INFO]: Updated /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/package.json
> [INFO]: Updated /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/opensearch_dashboards.json
> [ERROR]: changelog file is not defined. Use --manifest-changelog <path/to/file>.
> ```
> 
> However, applying that option the error persists
> 
> ```
> ❯ node scripts/release/tag.js --plugins-directory ./plugins --manifest-changelog ./CHANGELOG.md   --version 4.7.1 --revision 02 --platform-version 2.9.0
> 
> [INFO][tag]: Version: 4.7.1
> [INFO][tag]: Revision: 02
> [INFO][tag]: Platform version: 2.9.0
> [WARN][tag]: Ensure the base branch is created in the remote and it has updated the files: CHANGELOG.md, unit tests files, API data files, etc... See RELEASING.md. It does not modify these files.
> [WARN][tag]: This script will prune the local branches and tags, bump the plugins, commit and push the tags to the remote repository, deleting any unpushed changes.
> Do you want to continue? [y/N] y
> [INFO][tag]: Run command: git checkout 4.7.1
> Switched to branch '4.7.1'
> [INFO][tag]: Switched to branch: 4.7.1
> [INFO][tag]: Run command: git fetch --prune --prune-tags --force
> [INFO][tag]: Pruned local branches and tags
> [INFO][tag]: Generating tag: v4.7.1-2.9.0...
> [INFO][bump]: Plugin directories from ./plugins: main
> [INFO][bump][main]: Version: 4.7.1
> [INFO][bump][main]: Revision: 02
> [INFO][bump][main]: Platform version: 2.9.0
> [INFO][bump][main]: Package manifest: /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/package.json
> [INFO][bump][main]: Plugin manifest: /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/opensearch_dashboards.json
> [INFO][bump][main]: This will update the manifest files: package and platform plugin.
> [INFO][bump][main]: Updated /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/package.json
> [INFO][bump][main]: Updated /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/opensearch_dashboards.json
> [INFO][bump][main]: Calling to specific bump script
> [INFO]: Version: 4.7.1
> [INFO]: Revision: 02
> [INFO]: Platform version: 2.9.0
> [INFO]: Package manifest: /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/package.json
> [INFO]: Plugin manifest: /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/opensearch_dashboards.json
> [INFO]: This will update the manifest files: package and platform plugin.
> [INFO]: Updated /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/package.json
> [INFO]: Updated /home/tostti/src/forks/wazuh-dashboard-plugins/plugins/main/opensearch_dashboards.json
> [ERROR]: changelog file is not defined. Use --manifest-changelog <path/to/file>.
> ```

I was researching and this error could be caused by the executed script of the checkout branch doesn't have the changes of this pull request. The `tag` script should be used from the same branch that will checkout. For example, if we want to create the tag for 4.7.1, we should run the tag script from that branch, if not, it could execute unwanted code. If we want the tag script can be run from another branch, then we must load the code before checkout the branch.

You can pass the error related to the `manifest-changelog` if you ensure you have a remote branch that includes the code of this pull request whose name is equal to the value specified in the `--version` option.

```sh
# Move the the branch of this pull request
git checkout enhancement/6091-adapt-enhance-releasing-tools
# Create a local branch from the branch that contains the changes in the tools
git checkout -b 4.7.1
# Push the 4.7.1 branch to the remote repository
git push origin 4.7.1
# Use the 4.7.1 branch to create the tag throuth the script. If the API spec of 4.7.1 fails, try to use 4.6.0. There was an error in the API spec 4.7.0+.
node scripts/release/tag.js --plugins-directory ./plugins --manifest-changelog ./CHANGELOG.md --version 4.7.1 --revision 02 --platform-version 2.8.0 --plugin-main-generate-api-data-spec https://raw.githubusercontent.com/wazuh/wazuh/4.7.1/api/api/spec/spec.yaml
```

For example, I could run successfully using the explained process for the `4.6.0` branch (I created the `4.6.0` from the branch that contains this source code):
```console
toni@manjaro-talos:~/wazuh/src/my-wazuh-kibana-app (4.6.0)$ node scripts/release/tag.js --plugins-directory ./plugins --manifest-changelog ./CHANGELOG.md --version 4.6.0 --revision 02 --platform-version 2.8.0 --plugin-main-generate-api-data-spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml
[INFO][tag]: Version: 4.6.0
[INFO][tag]: Revision: 02
[INFO][tag]: Platform version: 2.8.0
[WARN][tag]: Ensure the base branch is created in the remote and it has updated the files that are not modified by the bump process. See RELEASING.md.
[WARN][tag]: This script will prune the local branches and tags, bump the plugins, commit and push the tags to the remote repository, deleting any unpushed changes. It is required you have configurated the repository to sign the commit and tag creation.
Do you want to continue? [y/N] y
[INFO][tag]: Run command: git checkout 4.6.0
Ya en '4.6.0'
[INFO][tag]: Switched to branch: 4.6.0
[INFO][tag]: Run command: git fetch --prune --prune-tags --force
[INFO][tag]: Pruned local branches and tags
[INFO][tag]: Generating tag: v4.6.0-2.8.0...
[INFO][bump]: Plugin directories from ./plugins: main
[INFO][bump][main]: Version: 4.6.0
[INFO][bump][main]: Revision: 02
[INFO][bump][main]: Platform version: 2.8.0
[INFO][bump][main]: Package manifest: /home/toni/wazuh/src/my-wazuh-kibana-app/plugins/main/package.json
[INFO][bump][main]: Plugin manifest: /home/toni/wazuh/src/my-wazuh-kibana-app/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: This will update the manifest files: package and platform plugin.
[INFO][bump][main]: Updated /home/toni/wazuh/src/my-wazuh-kibana-app/plugins/main/package.json
[INFO][bump][main]: Updated /home/toni/wazuh/src/my-wazuh-kibana-app/plugins/main/opensearch_dashboards.json
[INFO][bump][main]: Calling to specific bump script
[INFO][bump][main][specific]: Run command: cd /home/toni/wazuh/src/my-wazuh-kibana-app/plugins/main && yarn generate:api-data --spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml
{
  displayConfiguration: true,
  displayExamples: false,
  displayHelp: false,
  format: 'plugin',
  output: 'file',
  spec: 'https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml',
  outputDirectory: '/home/toni/wazuh/src/my-wazuh-kibana-app/plugins/main/common/api-info'
}
yarn run v1.21.1
$ node scripts/generate-api-data.js --spec https://raw.githubusercontent.com/wazuh/wazuh/$(node -e "console.log(require('./package.json').version)")/api/api/spec/spec.yaml --output file --output-directory common/api-info --display-configuration --spec https://raw.githubusercontent.com/wazuh/wazuh/v4.6.0/api/api/spec/spec.yaml
File was created! Path: /home/toni/wazuh/src/my-wazuh-kibana-app/plugins/main/common/api-info/security-actions.json
File was created! Path: /home/toni/wazuh/src/my-wazuh-kibana-app/plugins/main/common/api-info/endpoints.json
Done in 2.43s.

[INFO][bump]: Updated ./CHANGELOG.md
[INFO][tag]: Run command: git diff --exit-code --no-patch;echo -n $?
Review the changes done with the bump. y (continue), d (list changes: git diff), f (list modified files: git diff --name-only), N/n (cancel) y
[INFO][tag]: There are changes to commit.
[INFO][tag]: Run command: git diff --name-only
CHANGELOG.md
plugins/main/common/api-info/endpoints.json
plugins/main/common/api-info/security-actions.json
plugins/main/opensearch_dashboards.json
plugins/main/package.json

[INFO][tag]: Run command: git commit -S -am "Bump v4.6.0-2.8.0"
[INFO][tag]: Commited
[INFO][tag]: Run command: git tag -s -a v4.6.0-2.8.0 -m "Wazuh plugins for Wazuh dashboard v4.6.0-2.8.0"
[INFO][tag]: Created tag: v4.6.0-2.8.0
[INFO][tag]: Run command: git push origin v4.6.0-2.8.0
To https://github.com/Desvelao/wazuh-kibana-app.git
 * [new tag]             v4.6.0-2.8.0 -> v4.6.0-2.8.0
[INFO][tag]: Pushed tag v4.6.0-2.8.0 to remote
[INFO][tag]: Run command: git reset --hard origin/4.6.0
[INFO][tag]: Undone changes.
[WARN][tag]: A commit was added to the tag, but the branch 4.6.0 was reset to the state of theremote.
```

For now, I will add a warning to the documentation in `RELEASING.md`.

Tostti commented:
Confirm that the script is running correctly with the considerations mentioned by @Desvelao in the previous message.

Tostti reviewed (approved):
### Test 🟢 
### CR 🟢 

LGTM
